### ISSUE TITLE
The validation split is held out but never evaluated: no trainer sets `eval_strategy`, so `val_split: 0.1` costs 10% of the data and `val_loss` stays empty

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#1223]
## What happens

With the default `data.val_split: 0.1`, the loader moves 10% of the rows into a validation split, and the trainers pass it as `eval_dataset`. No trainer sets `eval_strategy`, and Hugging Face's default is `"no"`, so `Trainer.evaluate()` is never called. Those rows are withheld from training and used for nothing. The validation-loss sinks added in #713 (metrics column, `TrainEvent.val_loss`, the display row) never receive a value on the transformers backend.

Minimal repro on `main` `99d553f5` (CPU, tiny model):

```python
import json
from pathlib import Path

import transformers
from typer.testing import CliRunner

from soup_cli.cli import app

seen = {"evaluate_calls": 0}
_train, _evaluate = transformers.Trainer.train, transformers.Trainer.evaluate


def train(self, *args, **kwargs):
    seen.update(train_rows=len(self.train_dataset), eval_rows=len(self.eval_dataset),
                eval_strategy=str(self.args.eval_strategy))
    out = _train(self, *args, **kwargs)
    seen["log entries with eval_loss"] = sum("eval_loss" in e for e in self.state.log_history)
    return out


def evaluate(self, *args, **kwargs):
    seen["evaluate_calls"] += 1
    return _evaluate(self, *args, **kwargs)


transformers.Trainer.train, transformers.Trainer.evaluate = train, evaluate

Path("chat.jsonl").write_text("\n".join(json.dumps({"messages": [
    {"role": "user", "content": f"question {i}"}, {"role": "assistant", "content": f"answer {i}"}]})
    for i in range(20)))
Path("sft.yaml").write_text(  # data.val_split is left at its default
    "base: hf-internal-testing/tiny-random-LlamaForCausalLM\ntask: sft\n"
    "data: {train: chat.jsonl, format: chatml, max_length: 64}\n"
    "training: {epochs: 2, batch_size: 2, gradient_accumulation_steps: 1, logging_steps: 1, "
    "quantization: none}\noutput: ./out\n")
res = CliRunner().invoke(app, ["train", "--config", "sft.yaml", "--yes"])
print("exit", res.exit_code, [line.strip() for line in res.output.splitlines() if "Loaded:" in line])
print(seen)
```

```
exit 0 ['Loaded: 18 train samples']
{'evaluate_calls': 0, 'train_rows': 18, 'eval_rows': 2, 'eval_strategy': 'IntervalStrategy.NO', 'log entries with eval_loss': 0}
```

Control: the same run, with the spy setting `args.eval_strategy = "steps"` and `eval_steps = 2` just before the real `train()`, which is what a fix would do:

| run | `evaluate()` calls | `eval_loss` log entries | tracker `metrics` rows (total / with `val_loss`) | "Val loss" shown |
|---|---|---|---|---|
| as shipped | 0 | 0 | 19 / 0 | no |
| evaluation scheduled (control) | 9 | 9 | 28 / 9 | yes |

So #713's sinks work end to end; the only missing piece is the schedule. The same holds on the other trainers checked (setup only, default `val_split`): DPO and KTO each receive a 2-row `eval_dataset` with `args.eval_strategy = IntervalStrategy.NO`.

## Why

SFT's `TrainingArguments` kwargs (`src/soup_cli/trainer/sft.py:895-921`) have no `eval_strategy` or `eval_steps`, yet the trainer is built with `"eval_dataset": eval_ds` (`sft.py:1034`). The split itself comes from `_finalize` (`src/soup_cli/data/loader.py:253-257`), applied whenever `val_split > 0`, and the schema default is 0.1 (`src/soup_cli/config/schema.py:401`). #713 made the callback read `eval_loss` (`src/soup_cli/monitoring/callback.py:257`), but its tests feed synthetic `on_log` calls, and nothing on the transformers path ever produces one. Sixteen trainer wrappers pass an `eval_dataset`, and none sets `eval_strategy`; the only mention of it in `src/` is `src/soup_cli/bench/train_run.py:137-138`, which sets it to `"no"`. Per #739, on `backend: mlx` the value comes from mlx-lm's own validation loop, so the same `soup.yaml` reports a validation loss on MLX and none on transformers.

## Fix path

**Wire it.** When the dataset has a non-empty validation split, schedule evaluation in every wrapper through one shared helper (the way `training_seed_kwargs` is shared), for example `eval_strategy="steps"` with `eval_steps` tied to `logging_steps` or `save_steps`, so the #713 sinks get values. The cost is the policy question. On GRPO, PPO and online DPO an evaluation means generation, and on the preference trainers it is a full forward over chosen and rejected. The default cadence and any per-task opt-out need a decision, and a config field (`training.eval_steps`, or `eval_strategy`) is probably wanted.

**Stop withholding** on any path that still does not evaluate: either do not split, or warn at load that N rows are held out and never used. A hard refusal does not fit here, because the split is the default and every existing config would stop loading. Changing the default `val_split` is also a behaviour change (it changes which rows train) and would have to be called out.

## Acceptance

- [ ] A test trains SFT on the tiny model with the default `val_split` and asserts that `Trainer.evaluate()` runs at least once, `eval_loss` appears in `log_history`, and the tracker stores non-null `val_loss` rows. It fails on `main` (0 calls).
- [ ] A parametrised test over the wrappers that receive a validation split (sft, pretrain, dpo, ipo, kto, orpo, simpo, bco, grpo, distill, classifier, embedding, reward_model, prm, moe_lora_routing, asr): whenever `trainer.eval_dataset` is non-empty, `trainer.args.eval_strategy` is not `"no"`, or the task is documented as not evaluating and the loader did not withhold rows for it.
- [ ] Control: `val_split: 0` gives no evaluation and no withheld rows.
- [ ] Variants: `eval_steps` smaller and larger than the run's total steps, `gradient_accumulation_steps > 1`, and `epochs > 1` with evaluation at epoch ends if that option is offered.
- [ ] Any new CLI output (the "held out" warning, the Val loss row) is asserted through an ANSI-strip + whitespace-collapse helper.
